### PR TITLE
add a133 support

### PR DESCRIPTION
**Pull request by baidxi**
add a133 support

### CODE REVIEW

File: soc_info.c [hunk 1]
```diff
@@ -536,6 +550,18 @@ soc_info_t soc_info_table[] = {
 		.sid_offset   = 0x200,
 		.sid_sections = generic_2k_sid_maps,
 		.watchdog     = &wd_h6_compat,
+	},{
+		.soc_id       = 0x1855, /* Allwinner A133 */
+		.name         = "A133",
+		.spl_addr     = 0x24000,
+		.scratch_addr = 0x25000,
+		.thunk_addr   = 0x43000, .thunk_size = 0x200,
+		.swap_buffers = a133_sram_swap_buffers,
```
Comment (apritzel) [inline soc_info.c:559]: With the above changes, you might not need a separate swap_buffers array, instead I ended up re-using v831_sram_swap_buffers, which just stores away the IRQ FEL stack. We might want to use a higher address, though, to get more usable SRAM.

**Review by apritzel — CHANGES_REQUESTED** [review]
Hi,

thanks for sending this! I had something similar here locally, so happy to test and review.

File: soc_info.c [hunk 2]
```diff
@@ -536,6 +550,18 @@ soc_info_t soc_info_table[] = {
 		.sid_offset   = 0x200,
 		.sid_sections = generic_2k_sid_maps,
 		.watchdog     = &wd_h6_compat,
+	},{
+		.soc_id       = 0x1855, /* Allwinner A133 */
+		.name         = "A133",
+		.spl_addr     = 0x24000,
```
[the remark this answers is not in the document]
Comment (baidxi) [reply soc_info.c:556]: start at 0x20000 only 4K is valid, at very early. so change to SRAM C location

[the remark this answers is not in the document]
Comment (apritzel) [reply soc_info.c:556]: What makes you think so? I dumped all the first 16KB, twice, then wrote something to 0x22000 (or 0x23000), and read that back, with the rest untouched. So everything from 0x20000 to 0x23fff is perfect SRAM, available early. This is also mentioned in the manual: "16K(support Byte operation, clock source is AHB1)". So there is no mentioning of this being sparse, or restricted, or borrowed to another peripheral.
If you tried writing something to 0x21000 and it crashed: This is due to this area being used by the BROM FEL routine (for the IRQ stack), you must not touch the area between 0x21000 and 0x21400. You mention this yourself in the patch, by backing up this part.
So do you have any other insight into why there would be only 4K available?
As mentioned before, we cannot easily change this location to our liking, as this has to match where the BROM loads to.

**Pull request closed without merging**